src/index.ts

```diff
@@ -207,25 +212,73 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
       throwOnFailedInitialization: true, // always use true here as underlying instance fetch is surrounded by try/catch
       skipInitialPoll: config.skipInitialRequest ?? false,
     };
+    EppoJSClient.instance.setConfigurationRequestParameters(requestConfiguration);
 
-    EppoJSClient.instance.setLogger(config.assignmentLogger);
+    // We have two at-bats for initialization: from the configuration store and from fetching
+    // We can resolve the initialization promise as soon as either one succeeds
+    let initFromConfigStoreError = undefined;
+    let initFromFetchError = undefined;
 
-    // default behavior is to use a LocalStorage-based assignment cache.
-    // this can be overridden after initialization.
-    EppoJSClient.instance.useCustomAssignmentCache(new LocalStorageAssignmentCache());
-    EppoJSClient.instance.setConfigurationRequestParameters(requestConfiguration);
-    await EppoJSClient.instance.fetchFlagConfigurations();
```

Comment:
Doing this here, "serially" with `await configurationStore.init()`, has two consequences:
1. If the cached configuration is successfully loaded, we still need to then wait for the fetch of the updated configuration before `init()` promise resolves and `initialized` is set to `true`.
2. If either loading from the cache or fetching hits an error, we will say we failed to initialize (and possibly rethrow) despite a configuration being successfully loaded